api/src/main/java/com/datatorrent/api/StorageAgent.java

```diff
@@ -94,4 +94,24 @@
     public void setApplicationAttributes(AttributeMap map);
   }
 
+  interface AsyncStorageAgent
+  {
+    /**
+     * Make checkpoint for given windowID final. i.e after this method returns,
+     * the checkpoint is accessible for recovery.
+     *
+     * @param operatorId
+     * @param windowId
+     * @throws IOException
+     */
+    void finalize(int operatorId, long windowId) throws IOException;
```

Comment:
The only places where async agent writes checkpoint immediately is while launching the application, where initial checkpoints are written by master. and while running tests. the name could be anything the intent of this method is the checkpoints are available for recovery after this method returns. If I rename this to saveImmediately, then it seems similar to save, but don't have same parameters as save.